### DISCUSSION TITLE
Database migrations and restore with filestore

### DISCUSSION THREAD

**Discussion by pleirb** [Tecnativa/doodba-copier-template#383] [Q&A] [answered]
First of all, congratulations to the group of experts that makes all this magic possible, very grateful to be able to make use of these utilities. I want to learn more and try to walk the right path. I've already done a few tests quite successfully and now intend to use this for migrations. In the past I have been able to migrate thanks to openupgrade, I managed to migrate a fairly large database from version 10 to 11 and then to 12. Now a few years have passed and I want to continue migrating but I want to take advantage of these new tools. At first I would like to know what practice is suggested to restore a database, at a certain point I was able to do it, but after several trial and error attempts. Then I wanted to reproduce the same thing and I couldn't. Then I saw that there are tools like click-odoo. What environment is recommended to avoid errors? I think I can solve the migration thing later. I have been using the following command:
`docker-compose run --rm odoo odoo -d devel -u all --stop-after-init`
Thank you very much !

**Comment by pedrobaeza** [comment, marked as answer]
You can dump your DB from the instance with:

```
docker-compose run --user root -l traefik.enable=false --rm -v /root/:/mig odoo bash -c 'pg_dump -Fc -v -U odoo -f /mig/db.dump $PGDATABASE'
```

and then restore it in other instance:

```
docker-compose run -l traefik.enable=false --user root --rm -v /root/:/mig odoo bash -c 'pg_restore --no-owner -d $PGDATABASE /mig/db.dump'
```

and you can also transfer the filestore copying/rsyncing the files in the associated volume (normally in /var/lib/docker/volumes/).

**Reply by pleirb** [reply]
Gracias @pedrobaeza 
In the end, inspired by your solution, I decided to reform the command to be able to use the dump.sql that comes in the ZIP backups that I have, leaving it like this to restore:

`docker-compose run -l traefik.enable=false --user odoo --rm -v /local/path/to/folder/with/sql/backing/:/mig odoo bash -c 'dropdb $PGDATABASE && createdb $PGDATABASE && psql -d $PGDATABASE < /mig/dump.sql'`

Because using the volume that docker generates in /var/lib/docker/volumes/ gives me permissions problems in the migration, something with the UID of the root user that docker uses maybe, I'm not sure.
And then I run the migration with the command I initially put in and everything works great. Which maybe it would be nice to add the traefik.enable=false thing too right? That's to isolate the process from external requests, right?
There really is a learning cost with this doodba, but it is worth the tour, it leaves a lot of extra knowledge that is not bad at all. I have set the goal of being able to use Doodba for everything, migrations, development and production. I study on Saturdays, which is when I can, but then I need to see what it's like for production and development. Thank you very much!

**Reply by pedrobaeza** [reply]
Yes, there's not only one way to do things, as the toolset under Doodba allows to access things through several ways.

`traefik.enable=false` is for avoiding to affect the production environment if you're running other things (like migrations) in the same host.
